src/sagemaker/config/config_schema.py

```diff
@@ -477,6 +491,33 @@ def _simple_path(*args: str):
                             TYPE: OBJECT,
                             ADDITIONAL_PROPERTIES: False,
                             PROPERTIES: {
+                                SESSION: {
+                                    TYPE: OBJECT,
+                                    ADDITIONAL_PROPERTIES: False,
+                                    PROPERTIES: {
+                                        S3_BUCKET: {
+                                            "description": "sets `default_bucket` of Session",
+                                            "$ref": "#/definitions/s3Bucket",
+                                        },
+                                        S3_OBJECT_KEY_PREFIX: {
+                                            "description": (
+                                                "sets `default_bucket_prefix` of Session"
+                                            ),
+                                            TYPE: "string",
+                                            # Regex based on
+                                            # https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html
+                                            # For now, the regex only allows the "safe characters"
+                                            # specified by S3. The regex can be loosened (but not
+                                            # tightened) in the future without introducing
+                                            # backward incompatibility.
+                                            "pattern": (
+                                                r"^[a-zA-Z0-9!_.*'()-]+(/[a-zA-Z0-9!_.*'()-]+)*$"
```

Comment:
is there a reason we start with this pattern (is this based on aws docs)? Can we allow all?

Reply:
Will think about allowing all. I didnt realize earlier that even something like ```//ghh/./gkjhkjhkj///hf/``` can be a valid prefix. I think the way the SDK currently generates S3 paths could break with a prefix like that though (will try it out)

Reply:
I'm fine with this. If we allow "all" now and later if we want to add a regex => then it is not a backwards compatible change.

Reply:
removed regex and handled the main edge cases I was worried about by refactoring the S3 utils